tests/trainer/connectors/test_callback_connector.py

```diff
@@ -0,0 +1,17 @@
+from unittest.mock import Mock
+
+from pytorch_lightning import Trainer
+from pytorch_lightning.callbacks import LearningRateMonitor, ModelCheckpoint, ProgressBar
+
+
+def test_checkpoint_callbacks_are_last(tmpdir):
+    """ Test that checkpoint callbacks always get moved to the end of the list, with preserved order. """
```

Comment:
Can we also check that the other callback states are preserved?

Reply:
added that test right below, thanks for the suggestion!